webrtc/RTCRtpTransceiver-setCodecPreferences.html

```diff
@@ -0,0 +1,136 @@
+    const capabilities1 = RTCRtpSender.getCapabilities('audio');
+    const capabilities2 = RTCRtpReceiver.getCapabilities('audio');
+    transceiver.setCodecPreferences([...capabilities1.codecs, ... capabilities2.codecs]);
+
+  }, `setCodecPreferences() with both sender receiver codecs combined should succeed`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    transceiver.setCodecPreferences([]);
+
+  }, `setCodecPreferences([]) should succeed`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const capabilities = RTCRtpSender.getCapabilities('audio');
+    const { codecs } = capabilities;
+
+    if(codecs.length >= 2) {
+      const tmp = codecs[0];
+      codecs[0] = codecs[1];
+      codecs[1] = tmp;
+    }
+
+    transceiver.setCodecPreferences(codecs);
+
+  }, `setCodecPreferences() with reordered codecs should succeed`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const capabilities = RTCRtpSender.getCapabilities('video');
+    assert_throws(() => transceiver.setCodecPreferences(capabilities.codecs));
+
+  }, `setCodecPreferences() on audio transceiver with codecs returned from getCapabilities('video') should throw InvalidAccessError`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const codecs = [{
+      mimeType: 'audio/piepiper',
+      clockRate: 2000,
+      channels: 2,
+      sdpFmtpLine: 'a=fmtp:98 0-15'
+    }];
+
+    assert_throws(() => transceiver.setCodecPreferences(codecs));
+
+  }, `setCodecPreferences() with user defined codec should throw InvalidAccessError`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const capabilities = RTCRtpSender.getCapabilities('audio');
+    const codecs = [
+      ...capabilities.codecs,
+      {
+        mimeType: 'audio/piepiper',
+        clockRate: 2000,
+        channels: 2,
+        sdpFmtpLine: 'a=fmtp:98 0-15'
+      }];
+
+    assert_throws(() => transceiver.setCodecPreferences(codecs));
+
+  }, `setCodecPreferences() with user defined codec together with codecs returned from getCapabilities() should throw InvalidAccessError`);
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const capabilities = RTCRtpSender.getCapabilities('audio');
+
+    const { codecs } = capabilities;
+    assert_greater_than(codecs.length, 0,
+      'Expect at least one codec available');
+
+    const [ codec ] = codecs;
+    const { channels=2 } = codec;
+    codec.channels = channels+1;
+
+    assert_throws(() => transceiver.setCodecPreferences(codecs));
+
+  }, `setCodecPreferences() with modified codecs returned from getCapabilities() should throw InvalidAccessError`);
```

Comment:
I wonder if it will allow the object to is modified, but the in a way that results in the correct value surfacing?

```js
const channels = codec.channels;
Object.defineProperty(codec, "channels", {
  get() {
    console.log("channels property was accessed");
    return channels;
  }
});
```

Of course, there isn't a single browser that supports all of the functionality at once, lol.